### PR TITLE
[TypeSpecValidation] Add tests for LinterRuleset

### PR DESCRIPTION
**Pull request by mikeharder**

### CODE REVIEW

File: eng/tools/typespec-validation/src/rules/linter-ruleset.ts [hunk 1]
```diff
@@ -16,12 +15,11 @@ export class LinterRulesetRule implements Rule {
     let stdOutput = "";
     let errorOutput = "";
 
-    const configFile = join(folder, "tspconfig.yaml");
-    const configText = await readFile(configFile, "utf8");
+    const configText = await host.readTspConfig(folder);
```
Comment (mikeharder) [inline eng/tools/typespec-validation/src/rules/linter-ruleset.ts:18]: Rules should only read config via `host`

File: eng/tools/typespec-validation/src/rules/linter-ruleset.ts [hunk 2]
```diff
@@ -16,12 +15,11 @@ export class LinterRulesetRule implements Rule {
...
     const config = yamlParse(configText);
 
     const rpFolder =
-      config.options?.["@azure-tools/typespec-autorest"]?.["azure-resource-provider-folder"];
+      config?.options?.["@azure-tools/typespec-autorest"]?.["azure-resource-provider-folder"];
```
Comment (mikeharder) [inline eng/tools/typespec-validation/src/rules/linter-ruleset.ts:22]: Handle empty config

File: eng/tools/typespec-validation/src/rules/linter-ruleset.ts [hunk 3]
```diff
@@ -35,7 +33,7 @@ export class LinterRulesetRule implements Rule {
     }
     stdOutput += `files: ${JSON.stringify(files)}\n`;
 
-    const linterExtends = config.linter?.extends;
+    const linterExtends = config?.linter?.extends;
```
Comment (mikeharder) [inline eng/tools/typespec-validation/src/rules/linter-ruleset.ts:36]: Handle empty config

**Review by weshaggard — APPROVED** [review]

**Pull request merged**